servicepulse/intro-failed-messages.md

```diff
@@ -1,72 +1,93 @@
+ * **Archive group** - Archives all messages contained in the group. [Learn more about archiving messages](/servicepulse/intro-archived-messages).
 
-If a message fails repeated retry attempts, an indication is added, including the number of times it has failed.
 
-![Repeated failure indication](images/failed-messages-repeated-failure.png 'width=500')
+#### All Messages tab and Failed Message group detail view
 
-NOTE: The number of retry attempts for a message can be significant if the handler for that message is not [idempotent](/nservicebus/concept-overview.md#idempotence). Any processing attempt that invokes logic that does not participate in the NServiceBus transactional processing will not be rolled back on processing failure.
+The "View messages" link in a failed message group entry and the "All messages" tab display the individual failed messages inside that failed group or all individual failed messages, respectively.
 
+Note: Working with individual messages can be useful for testing system fixes before deciding to do any potentially time consuming bulk operations.
+
+![Failed Messages Page](images/intro-failed-messages-failed-messages-page.png 'width=500')
+
+Both of these individual message list views allow for taking the actions on an individual message, on custom message selections or on all messages contained in the view. 
+
+##### Managing individual failed messages
+
+The following actions can be taken on each individual message:
+
+ * **Message Details:** For a given failed message, displays the message type, exception description, endpoint name and location, and failure timestamp.
+ * **StackTrace:** Displays the full .NET exception stacktrace.
+ * **Headers:** Displays a complete set of message headers.
+ * **Body:** Displays the serialized message body.
+ * **Copy Message Id:** Copies the failed message unique ID to the clipboard.
+ * **Open in ServiceInsight:** Launches [ServiceInsight](/serviceinsight/), focusing on the failed message for in-depth analysis of the failure causes. This only works if ServiceInsight is installed on the local machine.
 
-### Archiving Failed Messages
+### Archived Messages tab
 
 Failed messages that cannot be processed successfully (or should not be retried due to various application-specific reasons) can be archived.
 
 ![Failed Message Archive](images/failed-messages-archive.png 'width=500')
 
-Archiving in ServicePulse means that the failed messages are marked as "Archived". Its data is still available, but it is no longer displayed in the Failed Messages list in ServicePulse and is not counted by the Failed Messages indicator in the ServicePulse dashboard. It also will not appear in any failed message groups.
+[Learn more about archiving messages in ServicePulse](/servicepulse/intro-archived-messages).
```

Comment:
> [Learn more about archiving messages in ServicePulse](/servicepulse/intro-archived-messages).

Learn more about [archiving messages in ServicePulse](/servicepulse/intro-archived-messages).